starter_code/src/movies.js

```diff
@@ -1,12 +1,41 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
-
+function turnHoursToMinutes(e){
+var minutos = 0
+    return  movies.map(function (e){
+        if(e.duration.includes("h")){
+        minutos += parseInt(e.duration.split("h")[0])*60;
+    }
+    var minutos2 = 0
+    if(e.duration.includes("min")){
+        e.duration.substring(e.duration.indexOf("m"))
+      }
+})
+}
 
 // Get the average of all rates with 2 decimals 
-
-
+function ratesAverage(movies){
+      var average = movies.reduce(function(accumulator, e){
+        return accumulator +  parseFloat((e.rate)/movies.length);
+      }, 0)
+      return Math.round(average*100)/100;
+    }
 // Get the average of Drama Movies
 
+function soloDrama(arr){
+       var drama = arr.filter(function(movie){
+           return movie.genre.includes("Drama");
+       });
+       return drama;
+    }
+    function dramaMoviesRate(arr) {
+           var drama = soloDrama(arr);
+       
+           if (drama.length == 0) {
+               return undefined;
+           }
+           return ratesAverage(drama);
+      }
 
 // Order by time duration, in growing order
 
```

Comment:
// Order by time duration, in growing order
function orderByDuration(arr) {
  arr.sort(function (a, b) {
    if (a.duration === b.duration) {
      if (a.title > b.title) {
        return 1;
      }
      if (a.title < b.title) {
        return -1;
      }
      return 0;
    }
    return a.duration - b.duration;
  });
  return arr;
}

// How many movies did STEVEN SPIELBERG
function howManyMovies(arr) {
  if (arr.length === 0) return;
  var drama = arr.filter(function (elem) {
    return elem.genre.includes('Drama');
  });
  var spielbergDramas = drama.filter(function (elem) {
    return elem.director.includes('Spielberg');
  });
  return 'Steven Spielberg directed ' + spielbergDramas.length + ' drama movies!';
}

// Order by title and print the first 20 titles
function orderAlphabetically(arr) {
  var result = [];
  var orderedArr = arr.sort(function (a, b) {
    if (a.title > b.title) {
      return 1;
    }
    if (a.title < b.title) {
      return -1;
    }
    return 0;
  });
  var limit = 20;
  if (orderedArr.length < 20) {
    limit = orderedArr.length;
  }
  for (var i = 0; i < limit; i++) {
    result.push(orderedArr[i].title);
  }
  return result;
}
// console.log(orderAlphabetically(movies));

// Best yearly rate average

function bestYearAvg(movies) {
  if (movies.length === 0) { return undefined; }

  var disctintYears = [];
  var aux = [];

  movies.forEach(function (e) {
    if (!aux.includes(e.year)) {
      aux.push(e.year);
      disctintYears.push({ year: e.year, rate: 0, films: 0 });
    }
  });

  disctintYears.forEach(function (y) {
    movies.forEach(function (e) {
      if (e.year === y.year) {
        if (!e.rate) {
          return;
        }
        y.rate += parseFloat(e.rate);
        y.films += 1;
      }
    });
    y.rate = parseFloat(y.rate / y.films);
  });

  disctintYears.sort(function (a, b) {
    if (a.rate === b.rate) {
      return a.year - b.year;
    }
    return b.rate - a.rate;
  });

  console.log(disctintYears);

  return 'The best year was ' + disctintYears[0].year + ' with an average rate of ' + disctintYears[0].rate;
}